.github/gha_monitor/gha_monitor/__main__.py

```diff
@@ -0,0 +1,109 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
```

Comment:
General thought:
It would be good to add somewhere a writeup of what this is intended to do, from a high level. As-is, the comments in `main()` sketch out the intended flow, but it would be good to have that defined in one place so that future readers/maintainers don't need to dig through the code to figure out what this is meant to do.